torchgeo/datasets/seasonet.py

```diff
@@ -0,0 +1,456 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""SeasoNet dataset."""
+
+import os
+import random
+from typing import Callable, Collection, Iterable, Optional
+
+import matplotlib.patches as mpatches
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import rasterio
+import requests
```

Comment:
We don't yet have a direct dependency on requests, which is why mypy is failing. See below, it might be possible to remove this.